crypto/bn/build.info: define OPENSL_IA32_SSE2 globally when needed

levitte opened the pull request:
This required an extension of the DEFINE syntax, to allow `DEFINE[]=def`

`DEFINE[]` definitions end up pushed in `@{$config{defines}}` instead of
being added to the output file list of defines.  This allows for the
unusual case where we need something to be defined globally, so it
gets picked up by anything using `$(CPPFLAGS)`.

bernd-edlinger commented:
No not yet working.
```
        $(CC) $(LIB_CFLAGS) $(LIB_CPPFLAGS) -c -o $@ crypto/modes/ghash-x86.s
crypto/modes/ghash-x86.s: ../openssl/crypto/modes/asm/ghash-x86.pl
        CC="$(CC)" $(PERL) ../openssl/crypto/modes/asm/ghash-x86.pl $(PERLASM_SCHEME) $(LIB_CFLAGS) $(LIB_CPPFLAGS) $(PROCESSOR) $@
```

the asm module does not define gcm_*_clmul
because it is not invoked with  -DOPENSSL_IA32_SSE2
```
uildtest_c_asn1t-bin-buildtest_asn1t.d.tmp -MT test/buildtest_c_asn1t-bin-buildtest_asn1t.o -c -o test/buildtest_c_asn1t-bin-buildtest_asn1t.o test/buildtest_asn1t.c
crypto/modes/fips-dso-gcm128.o: In function `CRYPTO_gcm128_init':
gcm128.c:(.text+0xdd): undefined reference to `gcm_init_clmul'
gcm128.c:(.text+0xe3): undefined reference to `gcm_gmult_clmul'
gcm128.c:(.text+0xef): undefined reference to `gcm_ghash_clmul'
collect2: error: ld returned 1 exit status
make[1]: *** [providers/fips.so] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/home/ed/OPC/asm'
make: *** [all] Fehler 2
```

it has this nice gadget:
```
$sse2=0;
for (@ARGV) { $sse2=1 if (/-DOPENSSL_IA32_SSE2/); }
```
all of them have that, but mostly it is beinign just lost performance.

levitte commented:
That's odd. It should have ended up in LIB_CPPFLAGS... I think, but I will have another look to make sure.

bernd-edlinger commented:
`
LIB_CPPFLAGS=-DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"$(OPENSSLDIR)\"" -DENGINESDIR="\"$(ENGINESDIR)\"" -DMODULESDIR="\"$(MODULESDIR)\"" $(CNF_CPPFLAGS) $(CPPFLAGS)`
`CNF_CPPFLAGS=-DNDEBUG`
`CPPFLAGS=`
-DOPENSSL_IA32_SSE2 w/o using `$(CFLAGS)` or so is on all compile rules, but not on the 
assembler perl scripts.

levitte commented:
Gotcha. Will fix soon

levitte committed:
Configure: Allow 'DEFINE[]=def'

DEFINE[] definitions end up pushed in @{$config{defines}} instead of
being added to the output file list of defines.  This allows for the
unusual case where we need something to be defined globally, so it
gets picked up by anything using $(CPPFLAGS).

levitte committed:
crypto/bn/build.info: define OPENSL_IA32_SSE2 globally when needed

This is done conditionally.

levitte commented:
How about now?

bernd-edlinger reviewed (approved):
looks good now

levitte commented:
Merged.

ef57f7996b6a93fc0ab80ed7dc17ff7d549869ff Configure: Allow 'DEFINE[]=def'
00372d7551ccf41ddb4fe8f00d62923e4dc65574 crypto/bn/build.info: define OPENSL_IA32_SSE2 globally when needed
